### PR TITLE
Prevent Dataflow options in parameters

### PR DESCRIPTION
**Pull request by damondouglas**
This PR throws more descriptive errors when users configure [dataflow_flex_template_job](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataflow_flex_template_job#parameters) or [dataflow_job](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataflow_job#parameters) `parameters` with Dataflow options, as documented: `do not configure Dataflow options here in parameters`.

```release-note:enhancement
dataflow: made provider return more descriptive errors when the `parameters` field of `google_dataflow_flex_template_job` contains Dataflow options
```

### CODE REVIEW

**Comment by damondouglas** [comment]
I tried to fix the formatting in the go.tmpl but editing in it is like using the old PC notepad as the IDE.

**Comment by trodge** [comment]
As far as I can tell this won't make any changes to the provider until we switch over to using the new go-based mmv1. To make changes now you'd need to edit `mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb`

**Comment by damondouglas** [comment]
> As far as I can tell this won't make any changes to the provider until we switch over to using the new go-based mmv1. To make changes now you'd need to edit `mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb`

Hi @trodge Thank you for your help. I moved the changes over to the erb files.

File: mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb [hunk 1]
```diff
@@ -841,4 +814,64 @@ func dataflowFlexJobTypeParameterOverride(ename, pname string, d *schema.Resourc
...
+			result = append(result, r)
+			continue
+		}
+		i++
+		r = runes[i]
+		result = append(result, unicode.ToUpper(r))
+	}
+
+	return string(result)
+}
```
Comment (trodge) [inline mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb:851]: I'm finding this function a bit hard to read. Could we use [SnakeToPascalCase](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tpgresource/utils.go#L593-L599) from the `tpgresource` package instead?

**Review by trodge — CHANGES_REQUESTED** [review]
Overall LGTM, except for one minor readability issue.

File: mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb (hunk 1 above)
Comment (damondouglas) [reply mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb]: @trodge Thank you for the tip. Done :-)

**Comment by damondouglas** [comment]
@trodge All checks past finally :-). Thank you for all your help.

**Comment by damondouglas** [comment]
@trodge This is ready for your review. Thank you, again.

**Review by trodge — APPROVED** [review]

**Comment by shuyama1** [comment]
A couple of tests started to fail in the nightly integration tests. Those tests were skipped in VCR and therefore the failures are not caught by presubmit tests in this PR.

Newly Failing tests:
* TestAccDataflowFlexTemplateJob_FullUpdate
* TestAccDataflowFlexTemplateJob_enableStreamingEngine
* TestAccDataflowFlexTemplateJob_withProviderDefaultLabels

Need to revert this change for now.

**Pull request merged**